dev/examples.el

```diff
@@ -1259,6 +1259,19 @@ new list."
     (funcall (-lambda (a b) (+ a b)) 1 2) => 3
     (funcall (-lambda (a (b c)) (+ a b c)) 1 (list 2 3)) => 6)
 
+  (defexamples -defun
+    (progn
+      (-defun dash-plus ((x y))
+        (+ x y))
+
+      (-map #'dash-plus '((1 2) (3 4) (5 6)))) => '(3 7 11)
+
+    (progn
+      (-defun a+b ((&plist :a a :b b))
+        (+ a b))
+
+      (-map 'a+b '((:a 1 :b 2) (:a 3 :b 4) (:a 5 :b 6)))) => '(3 7 11))
+
```

Comment:
I don't know if it's worth shortening these examples to make them clearer, but keep in mind that the `create-docs.sh` script is going to flatten these examples into a single line.

Reply:
I updated the example to call the `-defun` directly.